File: lib-android/src/main/java/com/pranav/lib_android/code/formatter/Formatter.java
package com.pranav.lib_android.code.formatter;

import com.pranav.lib_android.util.ConcurrentUtil;
import org.eclipse.jdt.internal.formatter.DefaultCodeFormatterOptions;
import org.eclipse.jdt.internal.formatter.DefaultCodeFormatter;
import org.eclipse.text.edits.TextEdit;
import org.eclipse.jface.text.IDocument;
import org.eclipse.jface.text.Document;

public class Formatter {

	private String source;

	public Formatter(String source) {
		this.source = source;
	}

	public String format() {
		DefaultCodeFormatterOptions options = DefaultCodeFormatterOptions.getEclipseDefaultSettings();

		final DefaultCodeFormatter codeFormatter = new DefaultCodeFormatter(options);

		final TextEdit edit = codeFormatter.format(
			DefaultCodeFormatter.K_COMPILATION_UNIT,
			source,
			0, // starting index
			source.length(), // length
			0, // initial indentation
			System.lineSeparator() // line separator
		);

		final IDocument document = new Document(source);
		try {
			edit.apply(document);
		} catch (Exception e) {
			throw new IllegalStateException(e);
		}
		// return the formatted code
		return document.get();
	}
}

File: lib-android/src/main/java/com/pranav/lib_android/code/formatter/Formatter.kt
package com.pranav.lib_android.code.formatter

import com.pranav.lib_android.util.ConcurrentUtil
import org.eclipse.jdt.internal.formatter.DefaultCodeFormatterOptions
import org.eclipse.jdt.internal.formatter.DefaultCodeFormatter
import org.eclipse.jface.text.Document

class Formatter(javaSource: String) {
	
	val source: String
	
	init {
		source = javaSource
	}
	
	fun format(): String {
		val options = DefaultCodeFormatterOptions.getEclipseDefaultSettings()
		
		val codeFormatter = DefaultCodeFormatter(options)
		
		val edit = codeFormatter.format(
	    	DefaultCodeFormatter.K_COMPILATION_UNIT,
	    	source,
    		0, // starting index
    		source.length(), // length
    		0, // initial indentation
	    	System.lineSeparator() // line separator
		)
		
		val document = Document(source)
		ConcurrentUtil.execute({
			try {
				edit.apply(document)
			} catch (e: Exception) {
				throw IllegalStateException(e)
			}
		})
		// return the formatted code
		return document.get()
	}
}
